galsim/sed.py

```diff
@@ -651,29 +632,26 @@ def calculateFlux(self, bandpass):
 
         @param bandpass   A Bandpass object representing a filter, or None to compute the bolometric
                           flux.  For the bolometric flux the integration limits will be set to
-                          (0, infinity) unless overridden by non-`None` SED attributes `blue_limit`
-                          or `red_limit`.  Note that SEDs defined using `LookupTable`s automatically
-                          have `blue_limit` and `red_limit` set.
+                          (0, infinity), which implies that the SED needs to be evaluable over
+                          this entire range.
 
         @returns the flux through the bandpass.
         """
         if self.dimensionless:
             raise TypeError("Cannot calculate flux of dimensionless SED.")
         if bandpass is None: # do bolometric flux
-            if self.blue_limit is None:
-                blue_limit = 0.0
-            else:
-                blue_limit = self.blue_limit
-            if self.red_limit is None:
-                red_limit = 1.e11 # = infinity in int1d
-            else:
-                red_limit = self.red_limit
-            if len(self.wave_list) > 0:
-                return(np.trapz(self(self.wave_list), self.wave_list))
-            else:
-                return galsim.integ.int1d(self, blue_limit, red_limit)
+            from galsim.deprecated import depr
+            depr('Using calculateFlux(bandpass=None) to compute a bolometric flux', 1.5, '',
+                 "If you need this functionality, you can use a pseudo-bolometric Bandpass created "
+                 "with:  bp = Bandpass('1', blue_limit=sed.blue_limit, red_limit=sed.red_limit)")
+            bp = galsim.Bandpass('1', self.blue_limit, self.red_limit)
+            return self.calculateFlux(bp)
```

Comment:
This needs a test in test_deprecated.  Normally the existing test of this functionality would be moved over to there with the check_dep call.  But there wasn't one, so I guess you'll need to make something for this functionality.  Probably could make a simple sed that has an analytic bolometric flux and make sure it comes out right.

Reply:
Looks like the depr message should get the 'nm' edit too...